### PR TITLE
How aboout full-fledged SKU types?

### PR DESCRIPTION
**Pull request by bonkey**
PR to #748 to make it more visible and more independent.

And, I'm not sure about it, too.

### CODE REVIEW

File: AtlasSDK/AtlasSDKTests/APIGuestOrderTests.swift [hunk 1]
```diff
@@ -80,7 +80,7 @@ extension APIGuestOrderTests {
     }
 
     fileprivate var cartRequest: GuestCartRequest {
-        let cartItemRequest = CartItemRequest(sku: "AD541L009-G11", quantity: 1)
+        let cartItemRequest = CartItemRequest(variantSKU: "AD541L009-G11000S000", quantity: 1)!
```
Comment (bonkey) [inline AtlasSDK/AtlasSDKTests/APIGuestOrderTests.swift]: @hani-ibrahim This was partially a reason of this PR. I spotted it in the middle of playing with it. Not sure if it's reasonable enough to have it, but maybe it's not that bad, anyway.

**Comment by bonkey** [comment]
👍

File: AtlasSDK/AtlasSDK/Endpoints/GetArticlesEndpoint.swift [hunk 2]
```diff
@@ -17,10 +17,10 @@ struct GetArticleEndpoint: CatalogEndpoint {
             ])
     }
 
-    let sku: VariantSKU
+    let sku: ConfigSKU
```
Comment (hani-ibrahim) [inline AtlasSDK/AtlasSDK/Endpoints/GetArticlesEndpoint.swift]: now we have `ConfigSKU` and above it with about 10 lines `Config` 👹

File: AtlasSDK/AtlasSDK/Models/SKU.swift [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+}
+
+extension SKU {
+
+    var isValid: Bool {
+        return Self.pattern.isValid(string: value)
+    }
+
+    init(validate string: String?) throws {
```
Comment (hani-ibrahim) [inline AtlasSDK/AtlasSDK/Models/SKU.swift]: probably we don't have any optional SKU anymore?

Comment (hani-ibrahim) [inline AtlasSDK/AtlasSDK/Models/SKU.swift]: `validated` or `validate` ? as the output will be a valid SKU or nil

Comment (bonkey) [inline AtlasSDK/AtlasSDK/Models/SKU.swift]: So, let's call it `valid`

File: AtlasSDK/AtlasSDK/Endpoints/GetArticlesEndpoint.swift (hunk 2 above)
Comment (bonkey) [inline AtlasSDK/AtlasSDK/Endpoints/GetArticlesEndpoint.swift]: Don't understand.

**Pull request merged**